library/Booster/Pattern/Util.hs

```diff
@@ -199,18 +202,29 @@ filterTermSymbols check = cata $ \case
         let concatSym = concatSymbol $ KListMeta def
             elemSym = klistElementSymbol def
             unitSym = unitSymbol $ KListMeta def
-         in if null heads
-                then [unitSym | check unitSym]
-                else filter check [concatSym, elemSym] <> concat heads
+         in case heads of
```

Comment:
should this not apply to map as well? i.e. `KMap _ [x] Nothing` only needs to `check x`

Reply:
I was not sure but thinking about it, yes, it is the same logic as in sets. Will adapt this case as well.